src/details/ArborX_DetailsTreeTraversal.hpp

```diff
@@ -70,6 +70,11 @@ struct TreeTraversal<BVH, Predicates, Callback, SpatialPredicateTag>
     }
   }
 
+  TreeTraversal(BVH const &bvh, Callback const &callback)
+      : _bvh{bvh}
+      , _callback{callback}
+  {}
+
```

Comment:
Generally, I wonder if this should be the only constructor, and whether the `parallel_for` should live outside of the `TreeTraversal` (possibly in a free function).

Reply:
In an earlier version, I had a run` member functions but then dropped to minimize changes. If we want to allow users to call the search in this non-batched way, i.e., inside of kernels, we should also provide a better interface than through `ArborX::Details::TreeTraversal`. I could imagine deriving a public (non-detail) class from this one or a free function.

Reply:
Wonder if we should provide
```c++
class BVH {
template<typename Predicate, typename Callback>
KOKKOS_FUNCTION void query(Predicate const& predicate, Callback const& callback);
}
```
or
```c++
template<typename BVH, typename Predicate, typename Callback>
KOKKOS_FUNCTION void query(BVH const& bvh, Predicate const& predicate, Callback const& callback);
```